filecoin-proofs/src/api.rs

```diff
@@ -67,19 +65,93 @@ pub struct SealOutput {
     pub piece_inclusion_proofs: Vec<PieceInclusionProof<PedersenHasher>>,
 }
 
+#[derive(Clone, Debug)]
+pub struct GeneratePoStOutput {
+    pub proof: Vec<u8>,
+}
+
 /// Generates a proof-of-spacetime, returning and detected storage faults.
 /// Accepts as input a challenge seed, configuration struct, and a vector of
 /// sealed sector file-path plus CommR tuples.
-///
 pub fn generate_post(
     post_config: PoStConfig,
-    challenge_seed: ChallengeSeed,
+    challenge_seed: &ChallengeSeed,
     input_parts: Vec<(Option<String>, Commitment)>,
-) -> error::Result<GeneratePoStDynamicSectorsCountOutput> {
-    generate_post_dynamic(GeneratePoStDynamicSectorsCountInput {
-        post_config,
+    faults: &[u64],
+) -> error::Result<GeneratePoStOutput> {
+    let vanilla_params = post_setup_params(post_config);
+    let setup_params = compound_proof::SetupParams {
+        vanilla_params: &vanilla_params,
+        engine_params: &(*ENGINE_PARAMS),
+        partitions: None,
+    };
+    let commitments_all: Vec<PedersenDomain> = input_parts
+        .iter()
+        .map(|(_, comm_r)| {
+            bytes_into_fr::<Bls12>(comm_r)
+                .map(Into::into)
+                .map_err(|err| match err {
+                    Error::BadFrBytes => {
+                        format_err!("could not transform comm_r into Fr32: {:?}", err)
+                    }
+                    _ => err.into(),
+                })
+        })
+        .collect::<error::Result<_>>()?;
+
+    let sector_size = u64::from(PaddedBytesAmount::from(post_config));
+    let sector_count = commitments_all.len() as u64;
+
+    let pub_params: compound_proof::PublicParams<_, rational_post::RationalPoSt<PedersenHasher>> =
+        RationalPoStCompound::setup(&setup_params).expect("setup failed");
+
+    let challenges = rational_post::derive_challenges(
+        vanilla_params.challenges_count,
+        sector_size,
+        sector_count,
         challenge_seed,
-        input_parts,
+        &faults,
+    );
+
+    let commitments: Vec<_> = challenges
+        .iter()
+        .map(|c| commitments_all[c.sector as usize])
```

Comment:
We don't have any panic-recovery tooling yet, so this will unwind the stack and crash the Filecoin node. Until we build those tools, please avoid using panics.